chips/sam4l/src/adc.rs

```diff
@@ -273,32 +274,53 @@ impl hil::adc::Adc for Adc {
                 // as the CPU clock
                 pm::enable_clock(Clock::PBA(PBAClock::ADCIFE));
                 nvic::enable(nvic::NvicIdx::ADCIFE);
-
-                // turn on GCLK10. The Generic Clock is needed for the ADC to
-                // work, although I have no idea what it is needed for. Its
-                // speed does not seem to matter, so let's just peg it to the
-                // CPU clock
-                scif::generic_clock_enable(scif::GenericClock::GCLK10, scif::ClockSource::CLK_CPU);
-
-                // determine clock divider
-                // we need the ADC_CLK to be a maximum of 1.5 MHz in frequency,
-                // so we need to find the PRESCAL value that will make this
-                // happen.
-                // Formula: f(ADC_CLK) = f(CLK_CPU)/2^(N+2) <= 1.5 MHz
-                // and we solve for N
-                // becomes: N <= ceil(log_2(f(CLK_CPU)/1500000)) - 2
-                let cpu_frequency = pm::get_system_frequency();
-                let divisor = (cpu_frequency + (1500000 - 1)) / 1500000; // ceiling of division
-                clock_divisor = math::log_base_two(math::closest_power_of_two(divisor)) - 2;
-                clock_divisor = cmp::min(cmp::max(clock_divisor, 0), 7); // keep in bounds
-                self.adc_clk_freq.set(cpu_frequency / (1 << (clock_divisor + 2)));
+                if frequency <= 3600 {
+                    // RC oscillator
+                    self.cpu_clock.set(false);
+                    let max_freq:u32;
+                    if frequency <= 1000 {
+                        scif::generic_clock_enable(scif::GenericClock::GCLK10, scif::ClockSource::RC32K);
+                        max_freq = 1000;
+                    } else {
+                        scif::generic_clock_enable(scif::GenericClock::GCLK10, scif::ClockSource::RCSYS);
+                        max_freq = 3600;
```

Comment:
They were empirically observed to be 1/32 of the clock frequency and I could not figure out the significance of the 32 though I think I understand it now. The clock is prescaled by a factor in the range 4 to 128. In the current configuration conversion takes 7 clock cycles and in continuous mode this means we can serve STRIG events every 7+1 cycles. This means that the maximum frequency we can serve in continuous mode is ADC_CLK/(4*(7+1)). I believe this also answers your question about the correctness of continuous mode.